adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/CookieTokenStore.java

```diff
@@ -98,14 +98,29 @@ public static void setTokenCookie(KeycloakDeployment deployment, HttpFacade faca
         }
     }
 
-    public static void removeCookie(HttpFacade facade) {
-        String cookiePath = getContextPath(facade);
+    public static void removeCookie(KeycloakDeployment deployment, HttpFacade facade) {
+        String cookiePath = getCookiePath(deployment, facade);
         facade.getResponse().resetCookie(AdapterConstants.KEYCLOAK_ADAPTER_STATE_COOKIE, cookiePath);
     }
 
-    private static String getContextPath(HttpFacade facade) {
+    static String getCookiePath(KeycloakDeployment deployment, HttpFacade facade) {
+        if (deployment.getAdapterStateCookiePath().startsWith("/")) {
```

Comment:
This is going to throw NPE when adapterStateCookiePath is not set (even though in this case it is initialized in KeycloakDeployment, sanity checks are generally preferred since this can be changed in the future)